astropy/table/tests/test_operations.py

```diff
@@ -28,13 +28,23 @@
 from astropy.utils import metadata
 from astropy.utils.compat import NUMPY_LT_2_0
 from astropy.utils.compat.optional_deps import HAS_SCIPY
+from astropy.utils.masked import Masked
 from astropy.utils.metadata import MergeConflictError
 
 
 def sort_eq(list1, list2):
     return sorted(list1) == sorted(list2)
 
 
+def check_cols_equal(col1, col2):
+    """Check that col1 == col2, taking care of zero-length masked columns."""
+    assert type(col1) is type(col2) or (
+        isinstance(col1, Masked) and type(col1) is Masked(type(col2))
```

Comment:
I would expect an equality check function to have input symmetry: the result shouldn't change if col1 and col2 are swapped, should it ?

Reply:
You're right. It's not needed right now but bound to cause problems later, so I better fix it.